client/asset/eth/eth.go

```diff
@@ -535,165 +540,151 @@ func decodeCoinID(coinID []byte) (*coin, error) {
 
 	amountCoinID, ok := id.(*srveth.AmountCoinID)
 	if !ok {
-		return nil,
-			fmt.Errorf("coinID is expected to be an amount coin id")
+		return nil, fmt.Errorf("coinID is expected to be an amount coin id")
 	}
-
-	return &coin{
-		id: *amountCoinID,
-	}, nil
+	return &coin{*amountCoinID}, nil
 }
 
-// FundOrder selects coins for use in an order. The coins will be locked, and
-// will not be returned in subsequent calls to FundOrder or calculated in calls
-// to Available, unless they are unlocked with ReturnCoins.
-// In UTXO based coins, the returned []dex.Bytes contains the redeem scripts for the
-// selected coins, but since there are no redeem scripts in Ethereum, nil is returned.
-// Equal number of coins and redeem scripts must be returned.
+// TODO: "fund" a buy order by locking funds for the redeem transaction gas.
+// func (eth *ExchangeWallet) FundBuyOrder(ord *asset.Order) { do not use ord.Value, just MaxSwapCount }
```

Comment:
This doesn't need to be part of every asset's interface, just a method to assert for at runtime to see if it's necessary.

Reply:
I agree with doing this.

Reply:
I think this will work.